### ISSUE TITLE
SPFx webpart is not available to add on site page

### ISSUE THREAD

**Issue by JLRishe** [SharePoint/sp-dev-docs#7782]
### Target SharePoint environment

SharePoint Online

### What SharePoint development model, framework, SDK or API is this about?

💥 SharePoint Framework

### Developer environment

Windows

### What browser(s) / client(s) have you tested

- [ ] 💥 Internet Explorer
- [ ] 💥 Microsoft Edge
- [X] 💥 Google Chrome
- [X] 💥 FireFox
- [ ] 💥 Safari
- [ ] mobile (iOS/iPadOS)
- [ ] mobile (Android)
- [ ] not applicable
- [ ] other (enter in the "Additional environment details" area below)

### Additional environment details

- browser version
  - Chrome: 99.0.4844.51
  - Firefox: 97.0.1
- SPFx version
  - 1.13.1
- Node.js version
  - 14.18.1

### Describe the bug / error

For some time now, I have had an issue where I'm able to add my SPFx webpart to pages in certain sites, but not others, even though the app is properly installed in both cases. The problem is that the webpart does not show in the list of available webparts when I try to add it to a page.

Other people within my organization are seeing the same issue, so it doesn't seem to be specific to my browser/login.

I believe that I've been able to resolve this in the past by some combination of deleting/uninstalling/emptying the recycle bin/re-installing the app, but as of right now, I am seeing this issue on several sites and no amount of deleting and reinstalling seems to remedy the issue.

What's more, I have a site where the webpart is present and functioning on one Site Page, and I'm able to add another instance of the webpart to the same page (it shows in the list of available webparts). But if I go to a new Site Page ___on the same site___, the webpart is not in the list of available webparts.

That last part in particular makes me think that this is a problem/bug in SharePoint.

What can I do to troubleshoot this?

If I open the Chrome browser tools and search for the webpart's GUID,

- On the page that already has the webpart, I find the GUID:
  - In the webpart's own .js file
  - In the current .aspx page's source, within a call to `spModuleLoader.start()`
- On the page that does not yet have the webpart, I don't find it anywhere.

I also have tried refreshing the problem page with the Disable Cache box checked, but that didn't make any difference.

I'm attaching the .sppkg file in question (renamed extension to .txt), and am including some screenshots below.

[FormsViewer Extensions.txt](https://github.com/SharePoint/sp-dev-docs/files/8190045/FormsViewer.Extensions.txt)

**Page that already has the wepbart:**
![image](https://user-images.githubusercontent.com/6964029/156869222-4ea12491-b19d-41f7-9bdd-8e8692938bc8.png)

<br />
<br />

**New page on same site:**
![image](https://user-images.githubusercontent.com/6964029/156869359-716e2f56-6d4d-4a0e-8c31-3bdf488a249f.png)

### Steps to reproduce

As I've noted above, this issue occurs unpredictably with no discernible pattern, so this is not guaranteed to reproduce the issue, but this is what I'm doing:

1. Ensure that you have a site collection-scoped app catalog
   - I use site collection-scoped app catalogs in order to have a Dev build of the .sppkg in one site collection and a Ship build in others, so I'm mentioning that here in case it's relevant.
2. Upload the .sppkg to the site collection-scoped app catalog
   - Do not select the "Make this solution available to all sites" option
3. Install the **FormsViewer Extensions** app in any site within the site collection
4. Add a new Site Page to the same site
5. Click the + button to add a webpart and look for a webpart called **FormsViewer Extension Config**

### Expected behavior

EXPECTED: The webpart should be in the list of webparts and you should be able to add it to the page.

ACTUAL: The webpart is not in the list of webparts.

**Comment by patmill** [comment]
Hey @JLRishe, can you try something for me?  If you go to the new page and save it (without the part showing), then reload the page with an added query string ?sw=bypass does the webpart show up?

**Comment by JLRishe** [comment]
@patmill Thank you for your reply. Unfortunately, that doesn't seem to have any effect. If I add that, the page simply reloads with `sw=bypass` absent from the URL, and the webpart is still unavailable.

FWIW, I've tried adding it to the saved page and then going into edit mode, and I've also tried adding it after going into edit mode (both before and after the `Mode=Edit`: `?Mode=Edit&sw=bypass`, `?sw=bypass&Mode=Edit`).

In each case, the outcome is the same - `sw=edit` is stripped from the URL and the webpart remains unavailable.

**Comment by JLRishe** [comment]
Hi @patmill 

As mentioned above, I tried your suggestion, but didn't have any luck.

However, I think I have been able to resolve this just now.

I also think I may have figured out what caused this: as I mentioned above, we use site collection-scoped app catalogs for this app, but one of my colleagues who didn't know this uploaded the app to our central app catalog. As soon as I realized what happened, I deleted it from the central app catalog and cleared the recycle bin, but I think that may have caused some conflict that lingered long after I did that.

So I tried the following today, and it seems to have remedied the issue.

1. Bump the app version up and and rebuild
2. Delete app from site collection-scoped app catalog and empty recycle bin
3. Upload newly built app to site collection-scoped app catalog
4. Go to site that's having the issue -> Site Contents -> About App -> Get It (new version)

After that, I'm able to add the webpart to site pages again.

So ultimately, it seems that what caused this was our mishandling of central/site collection-scoped app catalogs, but I wonder if  it indicates some bug in SharePoint that could be fixed. Presumably, having the app present in the site collection app catalog should have been enough to be able to use it normally without having to rebuild/delete/re-add.
